### PR TITLE
Fixes #15655 - update hostgroups in ptable migration without validation

### PR DESCRIPTION
**Pull request by shlomizadok**

### CODE REVIEW

File: db/migrate/20150514114044_migrate_ptables_to_templates.rb [hunk 1]
```diff
@@ -35,16 +35,14 @@ def up
...
       new_ptable.save!
 
       Audit.where(:auditable_type => 'Ptable', :auditable_id => old_ptable.id).each do |audit|
         audit.update_attribute :auditable_id, new_ptable.id
       end
 
-      old_ptable.hosts.each do |host|
-        host.update_attribute :ptable_id, new_ptable.id
-      end
+      Host::Managed.where(:ptable_id => old_ptable.id).update_columns(:ptable_id => new_ptable.id)
```
Comment (ohadlevy) [inline db/migrate/20150514114044_migrate_ptables_to_templates.rb]: :+1: this should speed up migration time as well.

**Comment by ohadlevy** [comment]
@shlomizadok note the length of commit message is too long

**Comment by ohadlevy** [comment]
[test] again

**Comment by ohadlevy** [comment]
@shlomizadok nearly ready. can you please ensure that you add the same logic to the down migration? please try with something like: `rake db:migrate:redo VERSION=20150514114044`

**Comment by shlomizadok** [comment]
@ohadlevy - changed also the down. Thanks for catching this up.
Also note that `rake db:migrate:redo VERSION=20150514114044` does not work in develop branch as well :/

**Comment by shlomizadok** [comment]
Test failure seem unrelated?

**Comment by ohadlevy** [comment]
[test]

File: db/migrate/20150514114044_migrate_ptables_to_templates.rb [hunk 2]
```diff
@@ -35,16 +35,14 @@ def up
       new_ptable.os_family = old_ptable.os_family
       new_ptable.type = 'Ptable'
       new_ptable.operatingsystems = old_ptable.operatingsystems
-      new_ptable.hostgroups = old_ptable.hostgroups
       new_ptable.save!
 
       Audit.where(:auditable_type => 'Ptable', :auditable_id => old_ptable.id).each do |audit|
         audit.update_attribute :auditable_id, new_ptable.id
```
Comment (tbrisker) [inline db/migrate/20150514114044_migrate_ptables_to_templates.rb]: While we're touching this migration, we can also get rid of this loop using `update_all`.

**Comment by tbrisker** [comment]
Not crucial in my opinion since this is just an old migration, but we could DRY up the up/down repetitions with a function that takes two parameters (the ptable or ptable id) in a different order.

**Comment by shlomizadok** [comment]
@tbrisker - Thanks. DRYed and added Audit as well

File: db/migrate/20150514114044_migrate_ptables_to_templates.rb [hunk 3]
```diff
@@ -67,16 +60,9 @@ def down
...
       old_ptable.save!
-
-      Audit.where(:auditable_type => 'Ptable', :auditable_id => new_ptable.id).each do |audit|
-        audit.update_attribute :auditable_id, old_ptable.id
-      end
-
-      new_ptable.hosts.each do |host|
-        host.update_attribute :ptable_id, old_ptable.id
-      end
+
```
Comment (tbrisker) [inline db/migrate/20150514114044_migrate_ptables_to_templates.rb:64]: nitpick: rubocop is :rage: about extra white space here

**Comment by tbrisker** [comment]
LGTM, I'll fix the extra whitespace on merge.

**Comment by ohadlevy** [comment]
:+1:

**Comment by shlomizadok** [comment]
:+1: thanks @tbrisker @ohadlevy

**Comment by tbrisker** [comment]
Merged as cf37d1f470c6eb898b65f0355360684abbabde16 with a bit of explanation added to the commit message and extra whitespace removed. Thanks @shlomizadok !

**Pull request closed without merging**